### ISSUE TITLE
Changelog/versioning: investigate changelog and versioning alternatives

### ISSUE THREAD

**Issue by SleepAPI** [nerolis-lab/nerolis-lab#50]
### Details

Currently we use the commit history as the changelog, but we should probably switch to proper versioning and changelog generation. The changelog could be used both to show users what has been changed, but also for us to parse the history in case we need to see recent changes.

I'm suggesting we use [semver](https://semver.org/) as version standard, it seems to be the most common used versioning semantic. 

For the changelog generation we should probably utilize the fact we already have conventional commit standard in place. Many tools exist to utilize this for changelog generation. A big one I found recently is Google's [release please](https://github.com/googleapis/release-please). An alternative I've used previously is [this one](https://www.npmjs.com/package/commit-and-tag-version), but that might be less optimal since we have access to Github actions.

The point of this ticket is to investigate, find alternatives and discuss with other Sleep API authors. Then create follow-up tickets for the implementations.

**Comment by sethdw** [comment]
To continue a bit of the discussion from discord:

If there's some bonkers bug that means charizard's ings completely out of whack, you could end up with a commit history like:
```fix: Fix issue 1 with production calculator
fix: Workaround issue 2 with production calculator
fix: Correct incorrect database entry
docs: Write documentation to avoid issue 1 in future```

All of which are distinct changes that need be separated - but from a user perspective, the only thing they care about is:
```fix: Charizard now has correct ingredient calculations```

Essentially we should probably decide what audience the changelog is actually intended for before anyone goes and implements anything

**Comment by SleepAPI** [comment]
> Essentially we should probably decide what audience the changelog is actually intended for before anyone goes and implements anything

Absolutely, this ticket exists only to investigate, discuss, decide and create follow-up tickets. It would be nice if we could generate one end-user changelog for the site and one technical for developers. If not, then we should look at striking a balance by configuring the generation to only trigger on certain commits.
